serialization/include/rp_ser.h

```diff
@@ -0,0 +1,361 @@
+ */
+void rp_ser_uninit(struct rp_ser *rp);
+
+/**@brief Function for sending command(function call) to the Remote processor.
+ *
+ * This function send procedure call to the Remote processor and waiting for response
+ * the specific amount of time if rsp is not NULL. After receiving command response
+ * the response decoder is called and received data can be returned.
+ *
+ * @param[in] rp Remote Procedure Serialization instance.
+ * @param[in] encoder Remote Procedure instance encoder.
+ * @param[in] rsp Command response decoder. If not NULL, function waiting for response and
+ *                decode it using this handler.
+ *
+ * @retval RP_SUCCESS Command send was successful.
+ * @retval RP_ERROR_NULL A parameter was NULL.
+ * @retval RP_ERROR_INVALID_PARAM A serialization packet length was 0.
+ */
+rp_err_t rp_ser_cmd_send(struct rp_ser *rp,
+			 struct rp_ser_encoder *encoder,
+			 cmd_rsp_handler_t rsp);
+
+/**@brief Function for sending event to the Remote processor.
+ *
+ * This function send event to the Remote processor. Event is asynchronous and
+ * receiving it is not confirmed by the Remote processor. Event can be used in case
+ * when remote function call is asynchronous or the Remote Processor has it own event and
+ * it is needed to pass it to other processor.
+ *
+ * @param[in] rp Remote Procedure Serialization instance.
+ * @param[in] encoder Event encoder.
+ *
+ * @retval RP_SUCCESS Command send was successful.
+ * @retval RP_ERROR_NULL A parameter was NULL.
+ * @retval RP_ERROR_INVALID_PARAM A serialization packet length was 0.
+ */
+rp_err_t rp_ser_evt_send(struct rp_ser *rp, struct rp_ser_encoder *encoder);
+
+/**@brief Function for sending command response to the Remote processor.
+ *
+ * This function send command response after receiving command was processed.
+ *
+ * @param[in] rp Remote Procedure Serialization instance.
+ * @param[in] encoder Response encoder.
+ *
+ * @retval RP_SUCCESS Command send was successfull.
+ * @retval RP_ERROR_NULL A parameter was NULL.
+ * @retval RP_ERROR_INVALID_PARAM A serialization packet length was 0.
+ */
+rp_err_t rp_ser_rsp_send(struct rp_ser *rp, struct rp_ser_encoder *encoder);
+
+/**@brief Function for initializing the remote procedure.
+ *
+ * This function initializes the Remote Procedure. Should be used
+ * after allocation of the buffer for encoding the Remote procedure.
+ *
+ * @param[in, out] encoder Remote Procedure encoder data.
+ * @param[in, out] container Serialized data container.
+ * @param[in] argc Serialized data count.
+ * @param[in] type Type of the Remote Procedure.
+ * @param[in] value Command/event number. In case of command response this
+ *                  parameter is not used.
+ *
+ * @retval RP_SUCCESS The Remote Procedure initialization was successful.
+ * @retval RP_ERROR_NULL A parameter was NULL.
+ * @retval RP_ERROR_INTERNAL Serializator encoding error.
+ */
+rp_err_t rp_ser_procedure_initialize(struct rp_ser_encoder *encoder,
```

Comment:
Can we call this function shorter e.g. _rp_ser_procedure_init()_ or perhaps _rp_ser_procedure_begin()_ to be consistent with _rp_ser_procedure_end()_?